### PR TITLE
Restore old time management

### PR DESCRIPTION
**Pull request by CounterPly**
Revert to old time management. No other functional changes.

https://github.com/official-stockfish/Stockfish/commit/aa88261a8f509fdabfe235042de1c1ea7a39a399

### CODE REVIEW

**Comment by lantonov** [comment]
Thanks for the pull request. I definitely didn't like the new TM. Do you mind a delay in committing to test it.

**Comment by MichaelB7** [comment]
New time management sucks in SD time control - e.g, game / 5 minutes

**Comment by CounterPly** [comment]
@lantonov No, not at all. Take your time. :)

**Comment by lantonov** [comment]
Some testing results so far
Bench 5051254 (not changed)
Score of asmFishW_2018-02-13_bmi2 (old_tm) vs asmFishW_b2e1185_bmi2 (master): +192 - 192 = 616 [0.500] 1000 games depth 14
Elo difference: 0.00 +/- 13.33
SPRT: llr 1.22e-13, lbound -2.94, ubound 2.94
0/500 different pairs

Speed comparison
Build Tester: 1.4.7.0
Windows 10 (Version 10.0, Build 0, 64-bit Edition)
Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
SafeMode: No 
Running In VM: No 
HyperThreading Enabled: No 
CPU Warmup: Yes
Command Line: bench 256 4 18 default depth
Tests per Build: 30
ANOVA: n/a
Engine 1: old_tm (test)
Engine 2: master
30 speed tests per build
```
                Engine# (NPS)                     Speedup     Sp     Conf. 95%    S.S.
2  (5,723,766.9   ) ---> 1  (5,657,435.5   ) --->  1.172%  98,027.3     Yes       No 
```
So there is some slowdown of 1.2% which is significant at 95% conf level
Some more stats: 
Best: 1 - 5,836,713 | 2 - 5,835,315
Worst: 1 - 5,299,695 |  2 -  5,577,962
Std Dev: 1 - 116,967.4 |  2 - 74,413.2  (the speed of the test build is more unstable)
A Linux version was tested on Bash on Ubuntu on Windows and it works:
Total time (ms) : 1876
Nodes searched  : 5051254
Nodes/second    : 2692566

**Comment by CounterPly** [comment]
Thanks for the update! The slowdown/destabilization you observe is interesting. Out of curiosity, do you observe a similar trend when you perform a speed comparison on the corresponding Stockfish versions?

**Comment by lantonov** [comment]
I have seen SD of the order 120-130k in other patches, too. Here, the bigger problem is that the wobble is towards the lower values. Max are almost equal (test even a bit greater) but the minimal values reached by test give it the stats significant slowdown

File: x86/CheckTime.asm [hunk 1]
```diff
@@ -1,86 +1,104 @@
-         calign   16
```
Comment (tthsqe12) [inline x86/CheckTime.asm:1]: An important function like CheckTime should be aligned to at least 16. It is actually not called that often, but it is speculatively executed by the processor and thus should be at least aligned to at least 16. Try keeping the align 16 or even upgrading it to align 64 and test the timings.

File: x86/CheckTime.asm [hunk 2]
```diff
@@ -1,86 +1,104 @@
...
-    ; The values of rdi at this point are in geometric progression.
-    ; On tested machine, this ends a 'go movetime 10000' in 10000 ms
-    ; with approx 130 calls to CheckTime.
+        ; rdi is target time
+               call   Os_GetTime
+                sub   rax, qword[time.startTime]
+                add   rax, 1
+        ; rax is elapsed time
+                sub   rdi, rax
+                 jb   .Stop
```
Comment (tthsqe12) [inline x86/CheckTime.asm:41]: This was upgraded to a "js" to handle possible timer wrap around. "js" is better here.

File: x86/Uci.asm [hunk 3]
```diff
@@ -1145,7 +1157,9 @@ UciBench:
            call  CmpString
            test  eax, eax
             jnz  .parse_hash
-            jmp  .parse_done
+           call  CmpString
+           test  eax, eax
+	        jmp .parse_done
```
Comment (tthsqe12) [inline x86/Uci.asm:1162]: The call and test do nothing here. The call to CmpString can also crash unless you know that rcx holds the address of a cstring.

**Comment by CounterPly** [comment]
@tthsqe12 

Thank you for these helpful critiques; I appreciate you taking the time to look through this.

**Comment by lantonov** [comment]
This is the speed test with the changes proposed by tthsqe12. 
Only 0.395% slowdown which is not significant.

Build Tester: 1.4.7.0
Windows 10 (Version 10.0, Build 0, 64-bit Edition)
Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
SafeMode: No 
Running In VM: No 
HyperThreading Enabled: No 
CPU Warmup: Yes
Command Line: bench 256 4 18 default depth
Tests per Build: 30
ANOVA: n/a
Engine 1: old_tm (test)
Engine 2: master
30 speed tests per build
```
                Engine# (NPS)                     Speedup     Sp     Conf. 95%    S.S.
2  (5,748,243.7   ) ---> 1  (5,725,624.2   ) --->  0.395%  102,195.3    No        No 
```
The slowdown of 0.395% is not significant at 95% conf level
Some more stats:
Best:       1 - 5,905,078 | 2 - 5,874,690
Worst:    1 - 5,277,365 | 2 - 5,461,620
Std Dev: 1 - 120,217.6 | 2 - 80,221.5

**Comment by lantonov** [comment]
Tonight I will make another speed test with align 64 and then merge the best in master

**Pull request closed without merging**